src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -988,35 +986,56 @@ public string Replace(char oldChar, char newChar)
             }
 
             // Copy the remaining characters, doing the replacement as we go.
-            ref ushort pSrc = ref Unsafe.Add(ref Unsafe.As<char, ushort>(ref _firstChar), copyLength);
-            ref ushort pDst = ref Unsafe.Add(ref Unsafe.As<char, ushort>(ref result._firstChar), copyLength);
+            ref ushort pSrc = ref Unsafe.Add(ref Unsafe.As<char, ushort>(ref _firstChar), (nint)(uint)copyLength);
+            ref ushort pDst = ref Unsafe.Add(ref Unsafe.As<char, ushort>(ref result._firstChar), (nint)(uint)copyLength);
+            nint i = 0;
 
             if (Vector.IsHardwareAccelerated && remainingLength >= Vector<ushort>.Count)
```

Comment:
Imagine a scenario where we are on an AVX2 machine and have `15` characters where index 8 is the first match. We `Buffer.Memmove` the first `8` (index 0-7, representing 16-bytes) then find we have `7` characters remaining and have to fallback to a scalar loop to handle it.

This effectively pessimizes the support added to "backtrack" so the "trailing" elements can be handled via vectorization.

It would likely be better to check that `Length >= Vector<ushort>.Count` so that we can backtrack and continue handling the trailing elements efficiently.

Reply:
Good idea 👍🏻 

> on an AVX2 machine and have `15` characters

I follow your idea (in general), but is `15` a mistake here? I mean `Vector<ushort>.Count` on AVX2 is `16`, so it would be an access violation to read 16 chars (ushorts) from the end of that string.

Reply:
Yeah, should be `31` and `16` rather than `15` and `8`. I was thinking I need to halve the bytes to get a count and then put numbers in the wrong spot 😅

Reply:
Done with 8627f6f4140af5bb08fd32a5e7abb3830b77a8a6